jaeger-context/src/main/java/com/uber/jaeger/context/TracingUtils.java

```diff
@@ -14,26 +14,37 @@
 
 package com.uber.jaeger.context;
 
-import io.opentracing.util.GlobalTracer;
-
+import io.opentracing.NoopTracer;
+import io.opentracing.NoopTracerFactory;
 import java.util.concurrent.ExecutorService;
 
+/**
+ * TracingUtils is going to be deprecated. To use its method please provide a tracer via
+ * {@link com.uber.jaeger.context.TracingUtils#setTracer(io.opentracing.Tracer)}.
+ */
+@Deprecated
 public class TracingUtils {
-  private static TraceContext traceContext = new ActiveSpanSourceTraceContext(GlobalTracer.get());
+  private static io.opentracing.Tracer tracer = NoopTracerFactory.create();
+  private static TraceContext traceContext;
+
+  public static void setTracer(io.opentracing.Tracer tracer) {
+    TracingUtils.tracer = tracer;
```

Comment:
maybe I am old school, but I was never comfortable with an assumption that assigning / accessing a reference is an atomic operation. I suggest adding `synchronized` to all public methods in this class.

Reply:
that's a very good point - and seems to be more solid design-wise 👍

Reply:
i think this applies to `com.uber.jaeger.filters.jaxrs2.TracingUtils` as well, apart from `com.uber.jaeger.context.TracingUtils`

Reply:
It is an assumption that the JVM adheres to the Java Language Spec.
https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.7

Could you elaborate on why one should be skeptical of a fundamental item in the language spec?